### DISCUSSION TITLE
Secure code (from development stage to production stage)

### DISCUSSION THREAD

**Discussion by d-solis-tovar** [micropython/micropython#12329] [ESP32]
Hi,

I developed some Micropython code for my esp32 project. Once developed and before commercial launch, I would like to find ways to secure this code (preventing users access). As I understand, the options available are:

- Convert .py files to .mpy. Users can access the file system but they would need to decompile the files to be able to read the content
- Convert .py files to .mpy and disable REPL
- Build my own Micropython firmware

I don't know if there are any additional options. I would like to know which approach you recommend me.

Thanks in advance!

**Comment by andrewleech** [comment]
There's some discussion about this topic underway here: https://github.com/orgs/micropython/discussions/12303
